polyfill/lib/ecmascript.mjs

```diff
@@ -310,16 +310,20 @@ export const ES = ObjectAssign({}, ES2020, {
     };
   },
   ParseTemporalInstantString: (isoString) => {
-    return ES.ParseISODateTime(isoString, { zoneRequired: true });
+    const result = ES.ParseISODateTime(isoString);
+    if (!result || (!result.z && !result.offset)) throw new RangeError('Temporal.Instant requires a time zone offset');
```

Comment:
Do we need `!result` here? It seems like an object will always be returned. (Same comment goes for ParseTemporalZonedDateTimeString below)

Reply:
Whoops, must have been left over from a previous go at solving this problem with the same "return false" pattern that we're actually removing. I'll remove and then merge.